currency/manager.go

```diff
@@ -186,6 +177,43 @@ func (p *PairsManager) StoreFormat(a asset.Item, pFmt *PairFormat, config bool)
 	return nil
 }
 
+// GetFormat returns the pair format in a concurrent safe manner
+func (p *PairsManager) GetFormat(a asset.Item, r bool) (PairFormat, error) {
```

Comment:
Can you expand the `r` name to something else a little easier for somebody to understand? This also needs a test.
 
suggestion: Also can consolidate this down a little bit, example below, it calls p.Get just to verify asset in every case, would need to change the error to something more generic for this to work:

```go
	ps, err := p.Get(a)
	if err != nil {
		return EMPTYFORMAT, err
	}
	var outbound *PairFormat
	if p.UseGlobalFormat {
		outbound = p.ConfigFormat
		if r {
			outbound = p.RequestFormat
		}
	} else {
		outbound = ps.ConfigFormat
		if r {
			outbound = ps.RequestFormat
		}
	}
	if outbound == nil {
		return EMPTYFORMAT, fmt.Errorf("global: %v request %v: %w", p.UseGlobalFormat, r, ErrAssetConfigFormatIsNil)
	}
	return *outbound, nil

```

Reply:
💯 can do
I had the same thoughts myself, started to do exactly this, but then thought I should rein myself in.
You've given me carte blanche so I'll have at it.
I wanted to simplify the error, as well. Seemed like overkill.

Reply:
Fixed 907d016

Thanks for the suggested simplification.
I ensured that we don't try to get the asset if we're using Global, and I kept a much simpler (existing) error for nil, which should be an edge case anyway.